test/merge.js

```diff
@@ -176,6 +237,33 @@ test('should work on array properties', function (t) {
     t.end()
 })
 
+test('should work on array properties with clone option', function (t) {
+    var src = {
+        key1: ['one', 'three'],
+        key2: ['four']
+    }
+    var target = {
+        key1: ['one', 'two']
+    }
+
+    var expected = {
+        key1: ['one', 'two', 'three'],
+        key2: ['four']
+    }
+
+    t.deepEqual(target, {
+        key1: ['one', 'two']
+    })
+    var merged = merge(target, src, {clone: true})
+    t.deepEqual(merged, expected)
+    t.ok(Array.isArray(merge(target, src).key1))
+    t.ok(Array.isArray(merge(target, src).key2))
```

Comment:
I don't think these three assertions (`deepEqual`, and the two `isArray`s) are necessary for this test, we should have other tests covering that behavior.  The cloning assertions below are all we should be asserting here.